src/app/record/object-record/object-record.component.ts

```diff
@@ -0,0 +1,84 @@
+import { Component, OnInit, Input } from '@angular/core';
+import { JsonPatch } from '../../shared/interfaces';
+
+@Component({
+  selector: 'object-record',
+  templateUrl: './object-record.component.html',
+  styleUrls: ['./object-record.component.scss']
+})
+export class ObjectRecordComponent implements OnInit {
+  @Input() record: object;
+  @Input() diff?: JsonPatch[];
+  @Input() depth: number;
+  @Input() expanded: boolean;
+
+  keys: Array<string> = [];
+  expandObject = false;
+  diffType: string;
+  private isArray = false;
+  isExpandedMap = new Map<string,boolean>();
+  diffKeys = new Map<string,boolean>();
+
+  ngOnInit() {
+    this.keys = Object.keys(this.record);
+    if (this.diff) {
+      this.diff.forEach(item => {
+        if (item.path.length > this.depth) {
+          this.diffKeys.set(item.path[this.depth],true);
+          this.expand(item.path[this.depth]);
```

Comment:
in my understanding, you expand the items with diffs so can't you do in the child component so that each component decides itself if it should be expanded or not rather than parent deciding for all children.

should be `expanded = this.diff && this.diff.length > 0`